source/NVDAObjects/behaviors.py

```diff
@@ -475,6 +476,13 @@ def _get_caretMovementDetectionUsesEvents(self):
 		prompt to be read when quickly deleting text."""
 		return False
 
+	def event_textChange(self) -> None:
+		"""Fired when the text changes.
+		@note: Updates also braille.
+		"""
+		super().event_textChange()
+		braille.handler.handleUpdate(self)

```

Comment:
```suggestion
		# braille.handler.handleUpdate must be called from the main thread.
		queueHandler.queueFunction(
			queueHandler.eventQueue,
			braille.handler.handleUpdate,
			self,
			_immediate=True
		)
```

Reply:
Thank you. I have removed lock, changed code as you suggested and tested. It seems to work.